gfapi: Implement glfs_symlinkat()

vatsa287 opened the pull request:
Updates: https://github.com/gluster/glusterfs/issues/2717
Sponsored-By: iXsystems, Inc https://www.ixsystems.com/
Depends-On: https://github.com/gluster/glusterfs/pull/3541
Signed-off-by: Shree Vatsa N [vatsa@kadalu.tech](mailto:vatsa@kadalu.tech)

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/api/src/glfs.h b/api/src/glfs.h
index 177f38824..43a8521f4 100644
--- a/api/src/glfs.h
+++ b/api/src/glfs.h
@@ -840,8 +840,8 @@ glfs_symlink(glfs_t *fs, const char *oldpath, const char *newpath) __THROW
     GFAPI_PUBLIC(glfs_symlink, 3.4.0);
 
 int
-glfs_symlinkat(struct glfs_fd *glfd, const char *oldpath, const char *newpath) __THROW
-    GFAPI_PUBLIC(glfs_symlinkat, 11.0);
+glfs_symlinkat(struct glfs_fd *glfd, const char *oldpath,
+               const char *newpath) __THROW GFAPI_PUBLIC(glfs_symlinkat, 11.0);
 
 int
 glfs_readlink(glfs_t *fs, const char *path, char *buf, size_t bufsiz) __THROW
diff --git a/tests/basic/gfapi/gfapi-o-path.c b/tests/basic/gfapi/gfapi-o-path.c
index 4d5218219..e9fd5db85 100644
--- a/tests/basic/gfapi/gfapi-o-path.c
+++ b/tests/basic/gfapi/gfapi-o-path.c
@@ -106,9 +106,10 @@ main(int argc, char *argv[])
     ret = glfs_lstat(fs, filepath_symlinkat, &buf);
     VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_lstat", ret, out);
 
-    if (!S_ISLNK(buf.st_mode)){
+    if (!S_ISLNK(buf.st_mode)) {
         ret = -1;
-        VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_symlinkat operation failed", ret, out);
+        VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_symlinkat operation failed", ret,
+                                         out);
     }
 
     ret = 0;
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/api/src/glfs.h b/api/src/glfs.h
index 177f38824..43a8521f4 100644
--- a/api/src/glfs.h
+++ b/api/src/glfs.h
@@ -840,8 +840,8 @@ glfs_symlink(glfs_t *fs, const char *oldpath, const char *newpath) __THROW
     GFAPI_PUBLIC(glfs_symlink, 3.4.0);
 
 int
-glfs_symlinkat(struct glfs_fd *glfd, const char *oldpath, const char *newpath) __THROW
-    GFAPI_PUBLIC(glfs_symlinkat, 11.0);
+glfs_symlinkat(struct glfs_fd *glfd, const char *oldpath,
+               const char *newpath) __THROW GFAPI_PUBLIC(glfs_symlinkat, 11.0);
 
 int
 glfs_readlink(glfs_t *fs, const char *path, char *buf, size_t bufsiz) __THROW
diff --git a/tests/basic/gfapi/gfapi-o-path.c b/tests/basic/gfapi/gfapi-o-path.c
index f94abeb6e..e9fd5db85 100644
--- a/tests/basic/gfapi/gfapi-o-path.c
+++ b/tests/basic/gfapi/gfapi-o-path.c
@@ -106,7 +106,7 @@ main(int argc, char *argv[])
     ret = glfs_lstat(fs, filepath_symlinkat, &buf);
     VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_lstat", ret, out);
 
-    if (!S_ISLNK(buf.st_mode)){
+    if (!S_ISLNK(buf.st_mode)) {
         ret = -1;
         VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_symlinkat operation failed", ret,
                                          out);
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/api/src/glfs.h b/api/src/glfs.h
index c3a27bad8..11a5aace5 100644
--- a/api/src/glfs.h
+++ b/api/src/glfs.h
@@ -841,8 +841,8 @@ glfs_symlink(glfs_t *fs, const char *oldpath, const char *newpath) __THROW
     GFAPI_PUBLIC(glfs_symlink, 3.4.0);
 
 int
-glfs_symlinkat(struct glfs_fd *glfd, const char *oldpath, const char *newpath) __THROW
-    GFAPI_PUBLIC(glfs_symlinkat, 11.0);
+glfs_symlinkat(struct glfs_fd *glfd, const char *oldpath,
+               const char *newpath) __THROW GFAPI_PUBLIC(glfs_symlinkat, 11.0);
 
 int
 glfs_readlink(glfs_t *fs, const char *path, char *buf, size_t bufsiz) __THROW
diff --git a/tests/basic/gfapi/gfapi-o-path.c b/tests/basic/gfapi/gfapi-o-path.c
index cbc4b1230..669946f32 100644
--- a/tests/basic/gfapi/gfapi-o-path.c
+++ b/tests/basic/gfapi/gfapi-o-path.c
@@ -118,7 +118,7 @@ main(int argc, char *argv[])
     ret = glfs_lstat(fs, filepath_symlinkat, &buf);
     VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_lstat", ret, out);
 
-    if (!S_ISLNK(buf.st_mode)){
+    if (!S_ISLNK(buf.st_mode)) {
         ret = -1;
         VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_symlinkat operation failed", ret,
                                          out);
```

vatsa287 committed:
gfapi: Implement glfs_symlinkat

Updates: gluster#2717
Sponsored-By: iXsystems, Inc <https://www.ixsystems.com>
Depends-On: #3541
Signed-off-by: Shree Vatsa N <vatsa@kadalu.tech>
